container-core/src/main/java/com/yahoo/container/di/CloudSubscriber.java

```diff
@@ -29,10 +32,16 @@ public class CloudSubscriber  implements Subscriber {
     // if waitNextGeneration has not yet been called, -1 should be returned
     private long generation = -1L;
 
-    CloudSubscriber(String name, ConfigSource configSource, Set<ConfigKey<ConfigInstance>> keys) {
+    CloudSubscriber(ExecutorService executor, String name, ConfigSource configSource, Set<ConfigKey<ConfigInstance>> keys) {
         this.name = name;
         this.subscriber = new ConfigSubscriber(configSource);
-        keys.forEach(k -> handles.put(k, subscriber.subscribe(k.getConfigClass(), k.getConfigId())));
+        Map<ConfigKey<ConfigInstance>, Future<ConfigHandle<ConfigInstance>>> futureHandles = new HashMap<>();
+        keys.forEach(k -> futureHandles.put(k, executor.submit(() -> subscriber.subscribe(k.getConfigClass(), k.getConfigId()))));
+        futureHandles.forEach((k, f) -> {
+            try {
+                handles.put(k, f.get());
+            } catch (InterruptedException | ExecutionException e) {}
```

Comment:
Do we want to ignore any exceptions thrown by submitted task (`ExecutionException`)?

Reply:
Some unit test started failing if i tried catching and rethrowing.
Will investigate further to see what is the better solution.

Reply:
Even when unwrapping the cause from `ExecutionException`?

Reply:
Yes, then I got a test that never finished.